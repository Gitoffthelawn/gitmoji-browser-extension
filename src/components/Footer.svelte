<script>
  import { page } from '../stores'
</script>

<footer>
  <nav>
    <button
      on:click={() => {
        page.goTo('settings')
      }}
    >
      settings
    </button>
    <a
      href="https://github.com/johannchopin/gitmoji-browser-extension"
      target="_blank"
      class="github"
      rel="noreferrer"
    >
      GitHub
    </a>
  </nav>
</footer>

<style>
  footer {
    position: relative;
    z-index: 1;
  }

  :global(body.dark) footer {
    background-color: black;
  }

  a {
    color: black;
    text-decoration: none;
  }

  :global(body.dark) a,
  :global(body.dark) button {
    color: white;
  }

  nav {
    display: flex;
    justify-content: center;
    padding: 0.25em 0;
  }

  nav > *:not(:last-child):after {
    content: "â€¢";
    margin: 0 0.2em;
  }

  button {
    border: none;
    padding: 0;
    background-color: transparent;
    font-size: 1em;
    cursor: pointer;
  }

  .github {
    margin-bottom: -0.2em; /* Little trick to center the link verticaly */
  }
</style>
