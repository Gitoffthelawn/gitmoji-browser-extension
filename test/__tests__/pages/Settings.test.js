import { render } from '@testing-library/svelte'
import Footer from '../../../src/pages/Settings.svelte'

describe('<Settings />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(Footer)
  })

  it('should match snapshot', () => {
    const { container } = wrapper

    expect(container).toMatchSnapshot()
  })
})
