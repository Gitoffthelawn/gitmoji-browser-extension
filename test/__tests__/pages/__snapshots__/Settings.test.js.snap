// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> should match snapshot 1`] = `
<body>
  <div>
    <header>
      <h1>
        <button>
          <img
            alt="icon: arrow-left"
            class="icon arrow-left "
          />
        </button>
        
    Settings
      </h1>
    </header>
     
    <div
      class="settings"
    >
      <div
        class="setting center"
      >
        <div
          class="theme-switch"
        >
          <button
            class="light"
            type="button"
          >
            Light
          </button>
           
          <button
            class="dark"
            type="button"
          >
            Dark
          </button>
        </div>
      </div>
       
      <div
        class="setting"
      >
        <div
          class="checkbox"
        >
          <input
            type="checkbox"
          />
           
          <label />
        </div>
         
        <h2>
          Close extension after copying a gitmoji
        </h2>
      </div>
       
      <div
        class="setting"
      >
        <div
          class="checkbox"
        >
          <input
            type="checkbox"
          />
           
          <label />
        </div>
         
        <h2>
          Show description on hover or focus
        </h2>
      </div>
       
      <hr />
       
      <div
        class="setting"
      >
        <div
          class="checkbox"
        >
          <input
            type="checkbox"
          />
           
          <label />
        </div>
         
        <h2>
          Inject ðŸ”€ gitmoji in GitHub/GitLab pull-request commit's message
        </h2>
      </div>
       
      <div
        class="setting center"
      >
        <div
          class="merge-gitmoji-type-select"
        >
          <h3>
            Select which type of gitmoji you want to use (code/emoji):
          </h3>
           
          <select>
            <option
              value="emoji"
            >
              ðŸ”€ 
            </option>
            <option
              value="code"
            >
              :twisted_rightwards_arrows:
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</body>
`;
